File: android/src/main/kotlin/com/tcubedstudios/flutter_audio_recorder4/FlutterMethodCalls.kt
package com.tcubedstudios.flutter_audio_recorder4

enum class FlutterMethodCalls(var methodName: String) {

    HAS_PERMISSIONS("hasPermissions");

    companion object {
        fun String.toFlutterMethodCall() = FlutterMethodCalls.values().firstOrNull { it.methodName == this }
    }
}

File: ios/Classes/FlutterMethodCalls.swift
import Foundation

enum FlutterMethodCalls: CaseIterable {
    
    case HAS_PERMISSIONS
    
    var methodName: String {
        get {
            switch self {
            default: //HAS_PERMISSIONS
                return "hasPermissions"
            }
        }
    }
}

extension String? {
    func toFlutterMethodCall() -> String? {
        return FlutterMethodCalls.allCases.first { $0.methodName == self }?.methodName
    }
}
